File: 0189-rotate-array/0189-rotate-array.java
class Solution {
    public void rotate(int[] nums, int k) {
        k = k % nums.length;
        nums = reverse(nums, 0, nums.length - 1);
        nums = reverse(nums, 0, k - 1);
        nums = reverse(nums, k, nums.length - 1);

        // return nums;
    }

    int[] reverse(int[] nums, int l, int r) {

        while(l <= r) {
            int temp = nums[l];
            nums[l] = nums[r];
            nums[r] = temp;
            l++;
            r--;
        }
        
        return nums;
    }
}

File: 0189-rotate-array/0189-rotate-array.py
class Solution:
    
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """

        k = k % len(nums)
        left = 0
        right = len(nums) - 1

        nums = self.reverseN(nums, 0, len(nums) - 1)
        
        nums = self.reverseN(nums, 0, k - 1)

        nums = self.reverseN(nums, k, len(nums) - 1)

    def reverseN(self, nums, left, right):
            while left < right:
                nums[left], nums[right] = nums[right], nums[left]
                left += 1
                right -= 1
            return nums   
        
    
        